src/gsy_e/gsy_e_core/user_profile_handler.py

```diff
@@ -98,10 +102,14 @@ def get_first_week_from_profile(self, profile_uuid, current_timestamp) -> dict:
         """
         if not isinstance(profile_uuid, uuid.UUID):
             profile_uuid = uuid.UUID(profile_uuid)
-        first_datapoint_time = select(
+        first_datapoint = select(
             datapoint for datapoint in self.Profile_Database_ProfileTimeSeries
             if datapoint.profile_uuid == profile_uuid
-        ).order_by(lambda d: d.time).limit(1)[0].time
+        ).order_by(lambda d: d.time).limit(1)
+        if len(first_datapoint) == 0:
+            raise ProfileDBConnectionException(
```

Comment:
Solving this will definitely help us understand further issues better. But I was thinking, when a simulation fails because of an exception, why the UI is stuck in the initializing/running phase? Shouldn't it be updated to failed immediately? How can we achieve that? I'm suggesting to catch the exception on a higher layer and set the status of the simulation to failed and prevent watchdog from restarting it as well. What do you think?

Reply:
This is handled. If there is an error in the simulation, the error is send to the DB on gsy-web (`SimulationResults.error_output`, I already debugged a lot of issues using this entry) and the status is set to failed.
There are rare occasions when the error appears when the setup and the simulation is initialized. When this happens, the simulation is indeed stuck in initialising. I hope it makes sense.